apps/volontulo/views/__init__.py

```diff
@@ -143,23 +147,26 @@ def _save_userprofile():
             )
         return form
 
-    def _handle_user_avatar_upload():
+    def _handle_user_avatar(action='update'):
         u"""Handle image upload for user profile page."""
-        gallery_form = UserGalleryForm(request.POST, request.FILES)
-        if gallery_form.is_valid():
-            userprofile.clean_images()
-            gallery = gallery_form.save(commit=False)
-            gallery.userprofile = userprofile
-            # User can only change his avatar
-            gallery.is_avatar = True
-            gallery.save()
-            messages.success(request, u"Dodano grafikę")
+        if action == 'delete':
+            return request.user.userprofile.clean_images()
         else:
-            errors = '<br />'.join(gallery_form.errors)
-            messages.error(
-                request,
-                u"Problem w trakcie dodawania grafiki: {}".format(errors)
-            )
+            gallery_form = UserGalleryForm(request.POST, request.FILES)
+            if gallery_form.is_valid():
+                userprofile.clean_images()
+                gallery = gallery_form.save(commit=False)
+                gallery.userprofile = userprofile
+                # User can only change his avatar
+                gallery.is_avatar = True
```

Comment:
When? Changing model structure right now? When you touch any bad application (not saying it's one of them) you always touch bad code. If you always applied this theory in practice, you'd do nothing but refactoring in some projects.